goldenmaster/examples/mqttclient/main.cpp

```diff
@@ -0,0 +1,107 @@
+#include <iostream>
+#include "testbed2/generated/mqtt/manyparaminterfaceclient.h"
+#include "testbed2/generated/mqtt/nestedstruct1interfaceclient.h"
+#include "testbed2/generated/mqtt/nestedstruct2interfaceclient.h"
+#include "testbed2/generated/mqtt/nestedstruct3interfaceclient.h"
+#include "tb_enum/generated/mqtt/enuminterfaceclient.h"
+#include "tb_same1/generated/mqtt/samestruct1interfaceclient.h"
+#include "tb_same1/generated/mqtt/samestruct2interfaceclient.h"
+#include "tb_same1/generated/mqtt/sameenum1interfaceclient.h"
+#include "tb_same1/generated/mqtt/sameenum2interfaceclient.h"
+#include "tb_same2/generated/mqtt/samestruct1interfaceclient.h"
+#include "tb_same2/generated/mqtt/samestruct2interfaceclient.h"
+#include "tb_same2/generated/mqtt/sameenum1interfaceclient.h"
+#include "tb_same2/generated/mqtt/sameenum2interfaceclient.h"
+#include "tb_simple/generated/mqtt/voidinterfaceclient.h"
+#include "tb_simple/generated/mqtt/simpleinterfaceclient.h"
+#include "tb_simple/generated/mqtt/simplearrayinterfaceclient.h"
+#include "tb_simple/generated/mqtt/nopropertiesinterfaceclient.h"
+#include "tb_simple/generated/mqtt/nooperationsinterfaceclient.h"
+#include "tb_simple/generated/mqtt/nosignalsinterfaceclient.h"
+#include "testbed1/generated/mqtt/structinterfaceclient.h"
+#include "testbed1/generated/mqtt/structarrayinterfaceclient.h"
+#include "apigear/mqtt/mqttclient.h"
+#include "apigear/utilities/logger.h"
+#include <iostream>
+#include <sstream>
+#include <cstdlib>
+#include <random>
+
+using namespace Test;
+
+ApiGear::Utilities::WriteLogFunc getLogging(){
+
+    ApiGear::Utilities::WriteLogFunc logConsoleFunc = nullptr;
+    ApiGear::Utilities::LogLevel logLevel = ApiGear::Utilities::LogLevel::Warning;
+
+    // check whether logging level is set via env
+    if (const char* envLogLevel = std::getenv("LOG_LEVEL"))
+    {
+        int logLevelNumber = 255;
+        std::stringstream(envLogLevel) >> logLevelNumber;
+        logLevel = static_cast<ApiGear::Utilities::LogLevel>(logLevelNumber);
+    }
+
+    logConsoleFunc = ApiGear::Utilities::getConsoleLogFunc(logLevel);
+    // check whether logging was disabled
+    if (logLevel > ApiGear::Utilities::LogLevel::Error) {
+        logConsoleFunc = nullptr;
+    }
+
+    // set global log function
+    ApiGear::Utilities::setLog(logConsoleFunc);
+
+    return logConsoleFunc;
+}
+
+int main(){
+    auto logConsoleFunc = getLogging();
+    std::mt19937 randomNumberGenerator (std::random_device{}());
+    std::uniform_int_distribution<> distribution (0, 100000);
+
+    auto randomId = distribution(randomNumberGenerator);
+    // MQTT clients need to have unique identifiers
+    auto mqttclient = std::make_shared<ApiGear::MQTT::Client>("testClient"+std::to_string(randomId));
```

Comment:
I don't think the randomId is necessary here, "testClient" should be enough or even better "uniqueNameTestClient", same goes for server side. It saves 5lines of code and shows your intentions